### DISCUSSION TITLE
Question regarding Refrence Variable in Infinity

### DISCUSSION THREAD

**Discussion by janhavibhavsar30** [grafana/grafana-infinity-datasource#652] [Q&A]
Hello Sir,

I have been using the Infinity plugin on Grafana v9.3.1 and building a dashboard using the JSON API.

I have created 3 panels and I wish to connect them via a reference variable.
I have created a reference variable 'course_var' for the same, but I am unable to see the option to refer to the given variable.

I have tried a tutorial by you that gives insight into variables using Grafana InfluxDB. Is there any way you can guide me on how I can reference in Infinity. I have tried many ways, and I unable to achieve it. 

This holds an important part of my thesis project. Any guidance on this issue will be of great help.

Waiting for your assistance and guidance on the same.

Attached below is a screenshot of my UI.

Regards,
Janhavi Ashish Bhavsar

![image (4)](https://github.com/yesoreyeram/grafana-infinity-datasource/assets/106957869/caf2d5d1-d6b5-4aaf-84de-1dccfbaf924d)
![image (5)](https://github.com/yesoreyeram/grafana-infinity-datasource/assets/106957869/8f4e2f3b-c609-456f-9503-2426e4731935)

**Comment by yesoreyeram** [comment]
Where do you want to pass the variable? In URL / header / query parameter / URL body ? Or you want to do filter your data based on variable value? Can you explain with sample json?

**Reply by janhavibhavsar30** [reply]
Hello Sir, 

Thank you so much for your reply. 

I wish to connect my panels and make the dashboard interactive. 
I have created the variable 'course_var'

I wish to filter data based on the variable value. I saw a couple of tutorials on youtube using the variables for Infinity plugin, although,  I am unable to reference the variable on my end. 

Given below is screenshot of my Dashboard. I wish to filter the data on the following panels by using the variable 'course' which has been created. Currrently, when I select the values from the dropdown, no changes are affected. 

Panels with require referencing variable for filtering: 
Time Taken per assessment
Total number of right and wrong answers
Marks scored in each subject

![image](https://github.com/yesoreyeram/grafana-infinity-datasource/assets/106957869/25390542-406f-41c5-960a-90f2ce067b56)

**Reply by yesoreyeram** [reply]
Nice. What parameters your URL accept? If the URL accepts the filter as a param, we can pass it. Otherwise you need to filter in the client side . ( In that case, can you show me a sample API response and the infinity query editor )

**Reply by janhavibhavsar30** [reply]
![image](https://github.com/yesoreyeram/grafana-infinity-datasource/assets/106957869/ca9ece12-2e08-4466-b433-32034e6edbee)
![image](https://github.com/yesoreyeram/grafana-infinity-datasource/assets/106957869/51f13ed8-d30e-4fbf-8886-557f73b10fd9)


This is how I am pulling the data, my URL is not having any filters, I am adding filters to the data using transform, and the parsing options. 

Can you please explain on what do you mean by client side? I am relatively new to grafana, hence I am unable to get this. 

I have attached below my queries for Infinity plugin, and also my API when retreived on localhost
![image](https://github.com/yesoreyeram/grafana-infinity-datasource/assets/106957869/41cb0db6-1748-4909-a3ab-424964106962)

**Reply by janhavibhavsar30** [reply]
@yesoreyeram Hello Sir, any update regarding this? 

Request you to kindly let me know if any more information is required from my end. 

Waiting for your response. 

Thanks and Regards,
Janhavi

**Reply by rkalayn3** [reply]
I have the exact same question. I am experimenting infinity plugin using JSON API and very new to Grafana. Example I have defined a variable called "severity" with values "High,Medium,Low". I want to apply "Severity" as a global filter in my dashboard so that user can select the value (High/Medium/Low) and dashboard can display visualizations based on the value selected.

**Reply by janhavibhavsar30** [reply]
Hello Sir, @yesoreyeram 

Any update regarding the issue? It is a big issue and a turning point in my thesis. 

Would like to know if there is any way to reference the variable? 

Really hopeful for input from your end. 
Request you to please help me out with the referencing. 

Thanks And Regards, 
Janhavi

**Reply by yesoreyeram** [reply]
Hiya.. sorry for the delay. Can you check this one https://github.com/yesoreyeram/grafana-infinity-datasource/discussions/655#discussioncomment-6746469

**Reply by yesoreyeram** [reply]
and this one https://github.com/yesoreyeram/grafana-infinity-datasource/discussions/655#discussioncomment-6854340
